ss58-js/mod.test.ts

```diff
@@ -0,0 +1,111 @@
+import { assertEquals, assertThrows } from "../deps/std/testing/asserts.ts";
+import { decode, encode } from "./mod.ts";
+
+const ALICE_PUBLIC_KEY = "d43593c715fdd31c61141abd04a99fd6822c8558854ccde39a5684e7a56da27d";
+
+[
+  {
+    name: "ss58.encode should encode 8-bit prefix polkadot address",
+    args: [0, ALICE_PUBLIC_KEY],
+    expected: "15oF4uVJwmo4TdGW7VfQxNLavjCXviqxT9S1MgbjMNHr6Sp5",
+  },
+  {
+    name: "ss58.encode should encode 8-bit prefix substrate address",
+    args: [42, ALICE_PUBLIC_KEY],
+    expected: "5GrwvaEF5zXb26Fz9rcQpDWS57CtERHpNehXCPcNoHGKutQY",
+  },
+  {
+    name: "ss58.encode should encode 16-bit prefix aventus address",
+    args: [65, ALICE_PUBLIC_KEY],
+    expected: "cLxkfNUiCYsb57YLhTJdNVKxUTB1VTpeygYZNhYuFc83KrFy7", // cspell:disable-line
+  },
+].forEach(({ name, args, expected }) => {
+  Deno.test({
+    name,
+    fn: () => {
+      // @ts-ignore
```

Comment:
Let's avoid ts-ignoring in situations where we don't have to.

Instead of having an `args` props, let's have `prefix` and `publicKey` props, which we can supply directly to the `encode` fn.